pkg/kubernetes/status/readiness.go

```diff
@@ -0,0 +1,100 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	label "github.com/kudobuilder/kudo/pkg/util/kudo"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// IsReady computes instance readiness based on current state of underlying resources
+// currently readiness examines the following types: Pods, StatefulSets, Deployments, ReplicaSets and DaemonSets
+// Instance is considered ready if all the resources linked to this instance are also ready (healthy)
+func IsReady(i kudoapi.Instance, c client.Client) (ready bool, msg string, err error) {
+	resources, err := healthResources(c, i.Name, i.Namespace)
+	if err != nil {
+		return false, "", err
+	}
+	ready = true
+	readinessMessage := ""
+	for _, res := range resources {
+		healthy, _, err := IsHealthy(res)
```

Comment:
`IsHealthy` will only return an error if it can't perform certain conversions to calculate health. The actual message regarding corresponding to the resource healthiness is provided by `msg`. While we should check for errors here, this should be done before checking `healthy`. If we want to add messages for healthy resources, use `msg` for that.